docs/Preprocessing-config.yml

```diff
@@ -0,0 +1,95 @@
+output_dir: '/home/Processed_temp_dir/'
+
+verbose: 10
+
+# This is the folder the data used for training is stored. The image stacks compose of label-free images, corresponding fluorescence images at different positions and z slices useful for training the virtual staining model.
+# The stacks must be processed to get the focus slice in the same positon in all z stack from different positions. If your images are not well aligned you can use the following [code](https://github.com/mehta-lab/microDL/blob/master/scripts/align_z_focus.py).
+
+input_dir: '/home/TrainingData_dir'
+
+# Specify all channels where basic normalization preprocessing is performed including label-free and fluorescence channels available in the image stacks.
+
+channel_ids: [0, 1, 2, 3]
+
+# Specify the slices which should be used for training. Some slices might be too out-of-focus' and thus may not yield any useful information for training or can even confuse the training process.
+# Pick the slices to be in good focus range, but diverse enough to get enough 3D information.
+
+slice_ids: [10, 11, 12, 13, 14, 15]
+
+# Chose the image positions which would be used for training.
+
+pos_ids: [3, 4, 5, 6, 7, 8, 9, 10, 11, 12]
+
+# Number of gpu workers used for the training compuation. This will depend on the computer specifications you use and the GPU.
+num_workers: 12
+
+# Normalize the images using the following parameters:
+normalize:
+
+# Specifies to normalize the dataset you are providing as input
+  normalize_im: 'dataset'
+
+# The fraction of pixels in your images which can be considered as foreground. min_fraction of 0.25 means 1/4th of the pixels in the image are foreground
+# if you are staining nucleus, image will be 25% covered by nuclear regions.
+  min_fraction: 0.25
+
+# Which channels are to be normalized : same length as length of channel id vector, specify if required or not using True or False
+  normalize_channels: [True, True, True, True]
+
+# Specify if the structures are uniform
+  uniform_struct: True
+
+# Mask parameters: defines segmnetion for foreground and background regions for staining
+
+masks:
+
+# which is your target channel, to be segmented for use in training the model? If you are virtually staining the nuclei, the DAPI channel will be defined
+  channels: [1]
+
+# The stride size for unimodal thresholding, or otsu thresholding
+  str_elem_radius: 3
+
+# method used for segmentation, can be unimodal if intensities are not uniform, otsu if signal is uniform
+  mask_type: 'unimodal'
+
+# Save format of the processed mask images (for visualization as well!)
+  mask_ext: '.png'
+
+
+# Tile parameters : Defines parameters for tile creation for training, from the different input channels and mask channel images
+tile:
+    
+# The parameter 'tile_size' defines the size of tiles to be created from input images useful for training the microDL models.
+
+  tile_size: [256, 256]
+
+# 'step_size' is the overlap between tiles
+
+  step_size: [128, 128]
```

Comment:
I believe step size is the stride between centers of tiles. It is only coincidentally also the overlap when it is half the tile size.